app/controllers/admin/deals_controller.rb

```diff
@@ -0,0 +1,52 @@
+module Admin
+  class DealsController < BaseController
+
+    before_action :find_deal_by_id, only: %i[show edit update destroy]
+
+    def index
+      @deals = Deal.order(created_at: :desc)
+    end
+
+    def show; end
+
+    def new
+      @deal = Deal.new
+    end
+
+    def edit; end
+
+    def create
+      @deal = Deal.new(deal_params)
+      if @deal.save
+        redirect_to admin_deals_path, info: t('.deal_saved')
+      else
+        render :new
+      end
+    end
+
+    def update
+      if @deal.update(deal_params)
+        redirect_to edit_admin_deal_path(@deal.id), success: t('.deal_updated')
```

Comment:
if successful then no need to send him to edit page, instead redirect to show page.